server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -522,6 +523,16 @@ private static void ensureRemoteStoreNodesCompatibility(DiscoveryNode joiningNod
             }
         } else {
             if (remoteStoreCompatibilityMode == CompatibilityMode.MIXED) {
+                if (joiningNode.getVersion().after(currentNodes.getMaxNodeVersion())) {
```

Comment:
Are there checks on min node version? Also please add a WARN logs

Reply:
Before entering MIXED mode, we ensure that all the nodes in the cluster are of the same version (#13080). Since we are not allowing higher versioned nodes and lower versioned nodes never join, we want all nodes to be of the same version in MIXED mode.
Sure, will add a warn log.